src/basic/REDUCE_STRUCT-Seq.cpp

```diff
@@ -0,0 +1,132 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-22, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "REDUCE_STRUCT.hpp"
+
+#include "RAJA/RAJA.hpp"
+
+#include <limits>
+#include <iostream>
+
+namespace rajaperf 
+{
+namespace basic
+{
+
+
+void REDUCE_STRUCT::runSeqVariant(VariantID vid, size_t RAJAPERF_UNUSED_ARG(tune_idx))
+{
+  const Index_type run_reps = getRunReps();
+  const Index_type ibegin = 0;
+  const Index_type iend = getActualProblemSize();
+
+  REDUCE_STRUCT_DATA_SETUP;
+
+  switch ( vid ) {
+
+    case Base_Seq : {
+
+      startTimer();
+      for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
+
+        Real_type xsum = 0.0; Real_type ysum = 0.0;
+        Real_type xmin = 0.0; Real_type ymin = 0.0;
+        Real_type xmax = 0.0; Real_type ymax = 0.0;
+
+        for (Index_type i = ibegin; i < iend; ++i ) {
+          REDUCE_STRUCT_BODY;
+        }
+
+        points.SetCenter(xsum/(points.N),ysum/(points.N));
+        points.SetXMin(xmin); points.SetXMax(xmax);
+        points.SetYMin(ymin); points.SetYMax(ymax);
+        m_points=points;
+
+      }
+      stopTimer();
+
+      break;
+    }
+
+#if defined(RUN_RAJA_SEQ)
+    case Lambda_Seq : {
+
+      auto reduce_struct_x_base_lam = [=](Index_type i) -> Real_type {
+                                   return points.x[i];
+                                 };
+
+      auto reduce_struct_y_base_lam = [=](Index_type i) -> Real_type {
+                                   return points.y[i];
+                                 };
+
+      startTimer();
+      for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
+
+        Real_type xsum = 0.0; Real_type ysum = 0.0;
+        Real_type xmin = 0.0; Real_type ymin = 0.0;
+        Real_type xmax = 0.0; Real_type ymax = 0.0;
+
+        for (Index_type i = ibegin; i < iend; ++i ) {
```

Comment:
I'm not sure if I agree with the lambda version here. I would think we should encapsulate the whole loop body in a single lambda, but then the trouble is the final values at the end. Perhaps we can capture those by reference? We could then reuse the `REDUCE_STRUCT_BODY` macro above